src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/code/RegisterAttributes.java

```diff
@@ -55,11 +55,11 @@ public RegisterAttributes(boolean isCallerSave, boolean isCalleeSave, boolean is
      * @return an array whose length is the max register number in {@code registers} plus 1. An
      *         element at index i holds the attributes of the register whose number is i.
      */
-    public static RegisterAttributes[] createMap(RegisterConfig registerConfig, RegisterArray registers) {
+    public static RegisterAttributes[] createMap(RegisterConfig registerConfig, List<Register> registers) {
```

Comment:
We should remove raw arrays as much as possible in JVMCI and replace them with immutable Lists.

Reply:
I have audited all the .clone() on array objects and changed as much as possible. Let me know if there is still some opportunity

Reply:
Looks good - thanks!